File: Offer/2264/main.go
package main

import "fmt"

func maxValue(grid [][]int) int {
	if len(grid) == 0 || len(grid[0]) == 0 {
		return 0
	}
	row, column := len(grid), len(grid[0])
	var i, j int
	for i = 1; i < row; i++ {
		grid[i][0] += grid[i-1][0]
	}
	for j = 1; j < column; j++ {
		grid[0][j] += grid[0][j-1]
	}
	for i = 1; i < row; i++ {
		for j = 1; j < column; j++ {
			grid[i][j] += max(grid[i-1][j], grid[i][j-1])
		}
	}
	fmt.Println(grid)
	return grid[row-1][column-1]
}

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}

File: Offer/2264/main.py
from typing import List


class Solution:
    def maxValue(self, grid: List[List[int]]) -> int:
        if len(grid) == 0 or len(grid[0]) == 0:
            return 0
        row, column = len(grid), len(grid[0])
        i, j = 0, 0
        for i in range(1, row):
            grid[i][0] += grid[i - 1][0]
        for j in range(1, column):
            grid[0][j] += grid[0][j - 1]
        for i in range(1, row):
            for j in range(1, column):
                grid[i][j] += max(grid[i - 1][j], grid[i][j - 1])
        return grid[-1][-1]
